### PR TITLE
maint: require py37, fix tests, use pyupgrade/isort/autoflake, use pyproject.toml, more f-strings

### PR DESCRIPTION
**Pull request by consideRatio**
I believed this project could use some love and care before I tried to figure out why jupyterhub-idle-culler concludes a server wasn't to be culled.

This is a general maintenance PR that I've done on various other projects, trying to get it up to date with various practices in the jupyterhub org.

---

I have another PR I'd like to make if this lands as well, making us depend on jupyterhub 1.3+ instead of supporting some unknown versions lower than 0.9 still. Since that would be breaking, I didn't want to make that part of this PR though.

### CODE REVIEW

**Comment by manics** [comment]
In https://github.com/jupyter-server/jupyter_server/issues/1186 @Carreau suggested f-strings should be avoided in logging.

**Comment by consideRatio** [comment]
Thanks @manics, I've opened https://github.com/jupyterhub/team-compass/issues/625 and will adjust this code base either before/after merge if we reach agreement to start doing so.

It sounds sensible to me!

**Comment by manics** [comment]
How about remove the f-strings commit from this PR? No need to block it

**Comment by consideRatio** [comment]
I just wanted to reduce the amount of work, having the other set of commits removing support for <0.9 and <1.3 that will get merge conflicts if i remove the f-strings.

**Comment by consideRatio** [comment]
Thank you for your review @manics!! To reduce your review workload, I'll revert the f-string changes to log statements in this PR!

I'm feeling a bit tired today, sorry for pushing back!

**Comment by consideRatio** [comment]
@manics @minrk I've now based on Min's comment https://github.com/jupyterhub/team-compass/issues/625#issuecomment-1437071634 removed my refactor to use f-strings in debug logging statements, but retained other f-string transitions.

File: RELEASE.md
Comment (yuvipanda) [inline RELEASE.md:9]: Is this a requirement for our other packages too? I don't believe i have that

**Review by yuvipanda — APPROVED** [review]
one question about conda-forge but otherwise LGTM!

File: RELEASE.md
Comment (minrk) [reply RELEASE.md:9]: No, conda-forge push rights shouldn't be necessary for any package. That process should be mostly automated and handled on the conda-forge side.

Comment (consideRatio) [reply RELEASE.md]: Thank you! Okay I've connected these in the past, but I'll disconnect these responsibilities.

In [bbb3818](https://github.com/jupyterhub/jupyterhub-idle-culler/pull/58/commits/bbb38187380914fce47fc7e09a64b4d56ac257e0) I fixed some mistakes I made when copying content from jupyterhub/jupyterhub, omitting links and a final step.

I still link out to conda-forge, but write:

>1. Following the release to PyPI, an automated PR should arrive within 24 hours
   to [conda-forge/jupyterhub-idle-culler-feedstock] with instructions on
   releasing to conda-forge. You are welcome to volunteer doing this, but aren't
   required as part of making this release to PyPI.

**Comment by yuvipanda** [comment]
Thanks @consideRatio

**Pull request merged**
